open-metadata-implementation/user-interfaces/ui-chassis/ui-chassis-spring/src/main/java/org/odpi/openmetadata/userinterface/uichassis/springboot/auth/LoggingRequestFilter.java

```diff
@@ -0,0 +1,50 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/* Copyright Contributors to the ODPi Egeria project. */
+package org.odpi.openmetadata.userinterface.uichassis.springboot.auth;
+
+import org.slf4j.MDC;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import java.io.IOException;
+import java.util.Collections;
+
+public class LoggingRequestFilter implements Filter {
+    private String path;
+
+    public LoggingRequestFilter(String path) {
+        this.path = path;
+    }
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain) throws IOException, ServletException {
+        HttpServletRequest httpRequest = (HttpServletRequest) servletRequest;
+        String requestURI = httpRequest.getRequestURI();
+        if (!path.equals(requestURI)) {
+            filterChain.doFilter(servletRequest, servletResponse);
+            return;
+        }
+        MDC.put("remoteAddress", httpRequest.getRemoteAddr());
```

Comment:
Do we have a concern here that this is potentially personally identifiable information -- the combination of host, username, session is quite specific. 

Can you elaborate on what is recorded for the logging. how this information is used and how it's control is documented?

Reply:
@planetf1 the purpose is to have audit trail logging to supervise all actions from all users.
It is security related.
There is no personal information involved.

Reply:
(oops skipped the thread - added comments in main body)